lte/gateway/python/scripts/user_trace_cli.py

```diff
@@ -0,0 +1,302 @@
+                ue_id["enb_ue_id"] = int(ue_id["enb_ue_id"], 16)
+
+            if ue_id and ue_id not in ue_id_list:
+                ue_id_list.append(ue_id)
+                mme_id_list.append(ue_id["mme_ue_id"])
+
+        # Some sessions may not have enodeb id, getting the mme_id from Attach request
+
+        attach_pattern = r"ATTACH REQ \(ue_id = (\w+)\) \(IMSI = " + imsi + r"\)"
+
+        for line in lines:
+            match = re.findall(attach_pattern, line)
+            if not match:
+                continue
+            mme_id = int(match[0], 16)
+            if mme_id not in mme_id_list:
+                mme_id_list.append(mme_id)
+                ue_id_list.append({"mme_ue_id": mme_id, "enb_ue_id": "None"})
+
+        print("\nIMSI: " + imsi + "\n")
+        print("{0:<15} {1:<15} ".format('mme_id', 'enodeb_id'))
+        print('-' * 25)
+
+        for v in ue_id_list:
+            mme, enodeb = v.items()
+            if enodeb[1] != "None":
+                print("{0} 0x{1:<10} {2} 0x{3:<15}".format(mme[1], format(mme[1], 'x'), enodeb[1], format(enodeb[1], 'x')))
+            else:
+                print("{0} 0x{1:<10} {2}".format(mme[1], format(mme[1], 'x'), enodeb[1]))
+
+        return ue_id_list
+
+    def get_session_trace(self, args):
+        """Provide a session trace given and IMSI and mme id(ue_id).
+        1. Get timestamps for ocurrences where the mme id (ue_id) was found. This will be "session_timestamps"
+        2. Filter out logs where the timestamp doesn't match the session_timestamps. This will be "time_filtered_logs"
+        3. Filter out logs where line doesn't match imsi or ue_id
+        4. From filtered logs, find and show if there are any known session errors.
+
+        Args:
+            args: args obtained from cli
+
+        Returns:
+            sorted_log: returned session trace sorted on timestamp
+        """
+
+        imsi = self.args.imsi
+        log_filter = {}
+        session_timestamps = self.get_session_timestamp(args)
+        time_filtered_logs = self.get_log_time_filtered(session_timestamps)
+
+        self.get_ue_id_pattern(args)
+        for line in time_filtered_logs:
+            match = re.findall(self.parttern_str, line, re.IGNORECASE)
+            match_imsi = re.findall(imsi, line)
+            if match or match_imsi:
+                match_time = re.findall(self.pattern_time, line)
+                log_filter[match_time[0][0]] = line
+
+        sorted_log = sorted(log_filter.items(), key=lambda kv: kv[1])
+
+        [print(value) for key, value in sorted_log]
+        self.get_session_error(sorted_log)
+        return sorted_log
+
+    def get_session_error(self, session_trace):
+        """
+        From the session trace, find if any of the logs match any know errors.
+        Provide the error found and suggestion to fix it.
+
+        Args:
+            session_trace: Filtered logs for a given imsi and mme id
+        """
+
+        errors_list = [
```

Comment:
move this on top of the file and define it as constant
```suggestion
        SESSIOND_ERRORS_LIST = [
```

Reply:
thanks for the suggestion!